packages/svelte/src/compiler/phases/1-parse/state/tag.js

```diff
@@ -270,55 +271,41 @@ function open(parser) {
 
 		parser.eat('(', true);
 
-		parser.allow_whitespace();
-
-		/** @type {import('estree').Pattern[]} */
-		const parameters = [];
-
-		while (!parser.match(')')) {
-			let pattern = read_pattern(parser, true);
+		let parentheses = 1;
+		let params = '';
 
-			parser.allow_whitespace();
-			if (parser.eat('=')) {
-				parser.allow_whitespace();
-				const right = read_expression(parser);
-				pattern = {
-					type: 'AssignmentPattern',
-					left: pattern,
-					right: right,
-					start: pattern.start,
-					end: right.end
-				};
-			}
-
-			parameters.push(pattern);
-
-			if (!parser.eat(',')) break;
-			parser.allow_whitespace();
+		while (!parser.match(')') || parentheses !== 1) {
+			if (parser.match('(')) parentheses++;
+			if (parser.match(')')) parentheses--;
+			params += parser.read(/^./);
```

Comment:
[not a new bug](https://svelte-5-preview.vercel.app/#H4sIAAAAAAAAE32NMQ7CMAxFr2LM0IWIPQIkzkE6RK0rIlonalJEsXx3GiYmxvf0n77gEEbKaG-C7CdCi9eU8IBlTRXyk8ZCG-e4zF01p50xEB9gzMWx7Ml3d3iBzyCwwhmadwOqcqxeHTv-7jmWf438NNvXFPswBOrRlnkhbfUDiwRjwKYAAAA=), but this logic probably won't be sufficient because these characters could appear inside strings etc. we should probably fix it both here and inside `read_pattern`...

Reply:
Uh right...would prevent the plus and minus between strings be sufficient? It should right?

Reply:
strings, comments and regular expressions